firefox-ios/Client/Frontend/Home/LegacyHomepageViewController.swift

```diff
@@ -959,6 +959,15 @@ extension LegacyHomepageViewController: HomepageViewModelDelegate {
                             category: .legacyHomepage)
         }
     }
+
+    func reloadViewIfNeeded() {
```

Comment:
Why do we need this additional reload logic? I am mostly asking because we don't do this on the new homescreen and is seems to work as expected?

Reply:
I added this just to avoid useless reload, since i see in the `HomePageViewModel` that it is tracked if the view needs reload or not.